drivers/include/mtd_sdmmc.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2017 HAW-Hamburg
+ *               2023 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_mtd_sdmmc mtd wrapper for sdmmc
+ * @ingroup     drivers_storage
+ * @brief       Driver for SD Memory Cards and MMCs/eMMCs using mtd interface
+ *
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for mtd_sdmmc driver
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ */
+
+#ifndef MTD_SDMMC_H
+#define MTD_SDMMC_H
+
+#include <stdint.h>
+
+#include "mtd.h"
+#include "sdmmc/sdmmc.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/**
+ * @brief   Device descriptor for a mtd_sdmmc device
+ *
+ * This is an extension of the @c mtd_dev_t struct
+ */
+typedef struct {
+    mtd_dev_t base;                    /**< inherit mtd_dev_t object */
+    sdmmc_dev_t *sdmmc;                /**< SD/MMC device descriptor */
+    uint8_t sdmmc_idx;                 /**< SD/MMC peripheral index  */
+} mtd_sdmmc_t;
+
```

Comment:
We might get rid of `sdmmc_idx` with some more (implementation specific) macro-magic: 

```C
#define SAM0_SDHC_MTD(name, conf)           \
static sam0_sdhc_dev_t name ## _dev =  {    \
        .sdmmc_dev = {                      \
                        .driver = &sam0_sdmmc_driver, \
                     },                     \
        .conf = conf,                       \
        .irq_wait = MUTEX_INIT_LOCKED,      \
    }                                       \
};                                          \
mtd_sdmmc_t name = {                        \
    .base = {                               \
       .driver = &mtd_sdmmc_driver,         \
    },                                      \
    .sdmmc = &name ## _dev,                 \
}
```

Reply:
I guess you would place it in the periperal drivers for each MCU? The problem is that `sam0_sdhc_dev_t`s wouldn't be defined if module `mtd` isn't used.

We use the `get_dev` approach also in other drivers, e.g. `periph_usbdev`.